src/include/detail/linalg/tdb_matrix.h

```diff
@@ -228,15 +253,17 @@ class tdbBlockedMatrix : public MatrixBase {
 
     /* The size of the array may not be the size of domain.  Use non-zero value
      * if set in constructor */
-    if (last_row_ == 0) {
-      last_row_ =
-          (row_domain.template domain<row_domain_type>().second -
-           row_domain.template domain<row_domain_type>().first + 1);
+    if (!last_row.has_value()) {
```

Comment:
This is where we used to test for zero -- now that we are passing an `std::optional`, for `first_row` and `first_col` we should also pass an optional to the constructor where we used to pass the zero sentinel (per the above comments).